### PR TITLE
Enables Docker plugin registration via plugin_server module

### PR DESCRIPTION
**Pull request by venilnoronha**
This PR moves the existing plugin registration logic from `build.bat` to the `plugin_server` module.

### CODE REVIEW

File: client_plugin/utils/plugin_server/plugin_server_windows.go [hunk 1]
```diff
@@ -26,7 +27,34 @@ import (
...
+
+// pluginConfig is the plugin config.
+var pluginConfig = fmt.Sprintf(`{ "Name": "vsphere", "Addr": "npipe://%s" }`, npipeAddr)
+
+// writePluginConfig writes the plugin config to the Docker plugins directory.
+func writePluginConfig() {
+	if err := os.MkdirAll(pluginsPath, 0755); err != nil {
+		log.WithFields(log.Fields{"path": pluginsPath,
+			"err": err}).Fatal("Failed to create plugin config directory ")
+		panic("Failed to create plugin config directory")
```
Comment (shaominchen) [inline client_plugin/utils/plugin_server/plugin_server_windows.go]: Failing with panic doesn't sound a good idea. I'd suggest to return err and fail in a consistent way as we did at line 82.

File: client_plugin/utils/plugin_server/plugin_server_windows.go [hunk 2]
```diff
@@ -26,7 +27,34 @@ import (
 	"github.com/docker/go-plugins-helpers/volume"
 )
 
-const npipeAddr = `\\.\pipe\vsphere-dvs` // Plugin's npipe address
+const (
+	// npipeAddr is the plugin's npipe address.
+	npipeAddr = "//./pipe/vsphere-dvs"
+
+	// pluginsPath is the Docker plugins directory path.
```
Comment (shaominchen) [inline client_plugin/utils/plugin_server/plugin_server_windows.go]: Nit: Docker plugins => Docker's plugin...

File: client_plugin/utils/plugin_server/plugin_server_windows.go (hunk 1 above)
Comment (venilnoronha) [inline client_plugin/utils/plugin_server/plugin_server_windows.go]: Addressed in f6ce550.

File: client_plugin/utils/plugin_server/plugin_server_windows.go (hunk 2 above)
Comment (venilnoronha) [inline client_plugin/utils/plugin_server/plugin_server_windows.go]: Addressed in f6ce550.

**Comment by venilnoronha** [comment]
@pshahzeb
1. As described, it moves the existing plugin registration logic from `build.bat` (which is now meant for only building the plugin on a Windows VM) to the `plugin_server` module (which would ensure that the plugin registration happens correctly every time the daemon starts). @shaominchen and I had a brief discussion on this a while back, and I missed to create an issue for this.

2. I've tested it manually, as we're still in the midst of porting e2e tests.

**Pull request merged**